src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py

```diff
@@ -934,15 +952,31 @@ def __call__(
                 latent_model_input = self.scheduler.scale_model_input(latent_model_input, t)
 
                 # controlnet(s) inference
+                if guess_mode and do_classifier_free_guidance:
```

Comment:
in terms of controlnet inputs, i.e. `latents`, `prompt_embeds` and `image`. is there any difference between these 2 configurations? 
1. not `do_classifier_free_guidance` 
2. `do_classifier_free_guidance` and `guess_mode`

Reply:
1 and 2 both share the commonality that they perform ControlNet only with the `conditional` condition of the prompt. Specifically, in the case of 2, the `latents` and `prompt_embeds` come in a concatenated state of both `unconditional` and `conditional` due to the processing above. Therefore, it is necessary to deliberately separate them and extract only the `conditional` conditions. Hence, this is the implemented.

As for the `image`, it is completely unrelated to the prompt for UNet, so there is no need to separate the processing under any conditions.